ansible_collections/arista/avd/plugins/module_utils/eos_designs_facts.py

```diff
@@ -1212,8 +1241,16 @@ def uplink_peers(self):
 
         These are used to generate the "avd_topology_peers" fact covering downlinks for all devices.
         '''
-        if self.uplinks is not None:
-            return [uplink['peer'] for uplink in self.uplinks]
+        fabric_name = get(self._hostvars, "fabric_name", required=True)
+        inventory_group = get(self._hostvars, f"groups.{fabric_name}", required=True)
```

Comment:
This was using self.uplinks before, where it did the same checks before adding an uplink. I moved it out to this function, to prevent generating all the uplinks (with "expensive" IP calculations) when just looking for the list of switches.